shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+    ///             changes.
+    ///
+    /// @param[in]  targeted_event      The object that contains both the
+    /// generated
+    ///                                 event and the event target.
+    /// @param[in]  bridge              The pointer to the accessibility bridge
+    /// that
+    ///                                 can be used for querying the
+    ///                                 accessibility information at the time
+    ///                                 the event is fired.
+    virtual void OnAccessibilityEvent(
+        AXEventGenerator::TargetedEvent targeted_event,
+        AccessibilityBridge* bridge) = 0;
+
+    //------------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    /// target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    /// action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(uint16_t target,
+                                             FlutterSemanticsAction action,
+                                             uint8_t* data,
+                                             size_t data_size) = 0;
+  };
+  //------------------------------------------------------------------------------
+  /// @brief      Creates a new instance of a accessibility bridge.
+  ///
+  /// @param[in]  user_data           A custom pointer to the data of your
+  ///                                 choice. This pointer can be retrieve later
+  ///                                 through GetUserData().
+  AccessibilityBridge(std::unique_ptr<AccessibilityBridgeDelegate> delegate,
+                      void* user_data);
+  ~AccessibilityBridge();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a semantics node update to the pending semantics update.
+  ///             Calling this method alone will NOT update the semantics tree.
+  ///             To flush the pending updates, call the CommitUpdates().
+  ///
+  /// @param[in]  node           A pointer to the semantics node update.
+  void AddFlutterSemanticsNodeUpdate(const FlutterSemanticsNode* node);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a custom semantics action update to the pending semantics
+  ///             update. Calling this method alone will NOT update the
+  ///             semantics tree. To flush the pending updates, call the
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
```

Comment:
shared_ptr?

Docs should explain when you'd use a custom action update instead of dispatching a semantics action above.

Reply:
the custom action here is the available custom action on a semantics node because the embedding sends the custom action information of a node separately from the node update. Not really sure how I should put in the doc here...

Reply:
You can provide examples of how this is used on different platform(s).

Reply:
previous comment https://github.com/flutter/engine/pull/23491/files#r556940310

I will expect it is used almost the same way for every platform